modules/analysis/src/service/load.rs

```diff
@@ -104,9 +104,11 @@ purl_ref AS (
     SELECT
         sbom_id,
         node_id,
-        array_agg(get_purl(qualified_purl_id)) AS purls
+        array_agg(qualified_purl.purl) AS purls
     FROM
         sbom_package_purl_ref
+    LEFT JOIN
+        qualified_purl ON (sbom_package_purl_ref.qualified_purl_id = qualified_purl.id)
```

Comment:
@ctron @JimFuller-RedHat Do either of you know how I can preserve this logic without either the `get_purl` proc or the stringified purl field?

This is the only place in the codebase that currently uses `get_purl`.

Reply:
I guess this has to be rewritten to join with sbom_package_purl_ref

leave that with me (As subject of a different PR)

Reply:
I can't finish this one without doing it myself, if we agree that's the way to go.

Reply:
I think you can use the recipe of the `cpe`, from the next few lines.

Reply:
ya agreed then ;)

Reply:
I can do that, but if we go that route, half of [these tests](https://github.com/trustification/trustify/blob/0a75ac4fcb56759876c0d4a22c48ab808fbf5bd8/modules/analysis/src/endpoints/tests/mod.rs#L399) will break (any referencing a purl), and I think prodsec is already relying on them.

I'd like to proceed with adding a `purl: String` field to `sbom_package_purl_ref` if y'all are ok with that.

Reply:
maybe I am getting confused ... and a little nervous - we want all the info in the purl, not restricted.

Reply:
Why would using the cpe pattern force us dropping any information?

Reply:
It doesn't. I'm conflating two issues, sorry. The cpe pattern works fine for removing `get_purl` here. I forgot that we currently use `get_purl` in [another place](https://github.com/trustification/trustify/blob/16d6066d7e2f4ad7ae8a6f54fe94b4171971c3bc/modules/analysis/src/service/load.rs#L295) -- that I'd already removed in this PR -- and removing that is what will break the tests mentioned above.